pkg/providers/aws/provider_redis.go

```diff
@@ -166,6 +179,84 @@ func (p *AWSRedisProvider) createElasticacheCluster(ctx context.Context, r *v1al
 	}}, croType.StatusMessage(fmt.Sprintf("creation successful, aws elasticache status is %s", *foundCache.Status)), nil
 }
 
+// Add Tags to AWS Elasticache
+func (p *AWSRedisProvider) TagElasticache(ctx context.Context, r *v1alpha1.Redis, stratCfg StrategyConfig, cacheClusterId string, preferredAvailabilityZone string) (types.StatusMessage, error) {
+	cacheSvcTag := elasticache.New(session.Must(session.NewSession(&aws.Config{Region: aws.String(stratCfg.Region)})))
+	cacheRegion := preferredAvailabilityZone
+	cacheRegion = cacheRegion[:(len(preferredAvailabilityZone))-1]
+	// get the account number
+	svc := sts.New(session.New())
+	inputSts := &sts.GetCallerIdentityInput{}
+	id, err := svc.GetCallerIdentity(inputSts)
+	if err != nil {
+		logrus.Error("Error GetCallerIdentity returned an error : s%", err)
+		id.Account = aws.String("")
+	}
+	accountId := *id.Account
+	// need arn in the following format arn:aws:elasticache:us-east-1:1234567890:cluster:my-mem-cluster
+	arn := "arn:aws:elasticache:" + cacheRegion + ":" + accountId + ":cluster:" + cacheClusterId
+
+	// Set the Tag values
+	defaultOrganizationTag, exists := os.LookupEnv("DEFAULT_ORGANIZATION_TAG")
+	if exists {
+		// Print the value of the environment variable
+		p.Logger.Infof("env var DEFAULT_ORGANIZATION_TAG : %s", defaultOrganizationTag)
+	}
+	for _, productName := range r.ObjectMeta.Labels {
+		clusterId, _ := resources.GetClusterId(ctx, p.Client)
+		cacheTag := []*elasticache.Tag{
+			{
+				Key:   aws.String(defaultOrganizationTag + "clusterId"),
+				Value: aws.String(clusterId),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "resource-type"),
+				Value: aws.String(r.Spec.Type),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "resource-name"),
+				Value: aws.String(r.Name),
+			},
+			{
+				Key:   aws.String(defaultOrganizationTag + "product-name"),
+				Value: aws.String(productName),
+			},
+		}
+		input := &elasticache.AddTagsToResourceInput{
+			ResourceName: aws.String(arn),
+			Tags:         cacheTag,
+		}
+		// add tags
+		_, err = cacheSvcTag.AddTagsToResource(input)
+		if err != nil {
+			msg := "Failed to add tags to AWS Elasticache :"
+			return types.StatusMessage(msg), err
+		}
+		//When snapshots are created add tags to them
+		inputDescirbe := &elasticache.DescribeSnapshotsInput{
+			CacheClusterId: aws.String(cacheClusterId),
+		}
+		snapshotList, _ := cacheSvcTag.DescribeSnapshots(inputDescirbe)
+		if snapshotList.Snapshots != nil {
+			for _, snapshot := range snapshotList.Snapshots {
+				snapshotArn := "arn:aws:elasticache:" + cacheRegion + ":" + accountId + ":snapshot:" + *snapshot.SnapshotName
+				logrus.Infof("Snapshots name : %s", *snapshot.SnapshotName)
```

Comment:
i think having a log in this block makes sense, however we could make it more verbose, for example

```
adding operator tags to snapshot <snapshot-name> (arn: <snapshot-arn>)
```

i also think there may be some ability to extract tagging out to a util function, however i will create a separate low priority ticket for this.

Reply:
@aidenkeating Originally I had the arn in the log, but removed it as it gives account name, I deemed this a security risk and removed it. Am I being too paranoid ?

Reply:
@austincunningham I think that's valid, let's just keep the snapshot name